content/references/rippled-api/api-conventions/basic-data-types.md

```diff
@@ -21,6 +21,24 @@ Each closed [Ledger](ledger-data-formats.html) has a [Ledger Index][] and a [Has
 {% include '_snippets/data_types/hash.md' %}
 <!--{#_ #}-->
 
+### Hash Prefixes
+[[Source]<br>](https://github.com/ripple/rippled/blob/master/src/ripple/protocol/HashPrefix.h "Source")
+
+In many cases, the XRP Ledger prefixes an object's binary data with a 4-byte code before calculating its hash, so that objects of different types have different hashes even if their binary formats are the same. The 4-byte codes are structured as three bytes of ASCII followed by a zero byte.
+
+The following table shows the hash prefixes for some common types of data:
```

Comment:
Do we have all the types vs. just some? i.e. can we just fill out this table to cover all the cases?

Reply:
I waffled on this a bit, trying to strike a balance between providing everything and not providing info that users will almost certainly not need.

In the end I kept finding more cases (like signing Claims for payment channels) where users might want to know a certain prefix I had thought they wouldn't, so I went with including everything.